### ISSUE TITLE
[Improvement] Add Web Notifications API in WebUI

### ISSUE THREAD

**Issue by Azef1** [shinken-monitoring/mod-webui#510]
Hi,
It's a suggestion : 
When the state of a service/host change, generate a notification with Web Notifications API from W3C.
This new API allow to alert the user outside of a web page by displaying notifications (that do not require interaction by the user). Some browsers are compatibles with this API [see](http://caniuse.com/#feat=notifications). Demo [here](https://davidwalsh.name/demo/notifications-api.php)
The users can subscribe or not to this feature.
This feature will be very cool for users who are often on their browser. Users doesn't need to check periodically Shinken WebUI.

What do you think ?

**Comment by mohierf** [comment]
Nice idea !

**Comment by maethor** [comment]
It doesn't seem easy to implement because we don't detect state changes, we only display them. This feature would need a change queue, something like that. I don't know how to implement this kind of stuff.

But it's a nice idead. Would be usefull, I agree.

**Comment by mohierf** [comment]
@maethor : this may be implemented with the brok management that is done in the `manage_brok_thread` function. When it receives an host/service/contact update brok, you should raise a notification

**Comment by maethor** [comment]
Yes, but I don't know how to route it to the webui. I'm not a web developer :smile:

**Comment by maethor** [comment]
I mean, how do you implement this?

Do you make an infinite loop in the frontend to check if there is notifications to display? If so, where do you store the notifications until you display them?

Do you push the notifications from the backend when manage_brok_thread detect something? How do you do that?

**Comment by maethor** [comment]
If I understand correctly, we should do it with websockets. Sadly I don't have time to learn it, so I won't do it.

**Comment by mohierf** [comment]
What I would do?

 For the server part, I would create a new thread function (copy of the existing one) to get the host/service status update broks. If an host/service status changed, I would store this information in a global notifications list. 

Then, for the Web part, the WebUI client periodically raises an Ajax call to */gotfirstdata* which may be used to pull information from the notifications list and raise a popp on the client browser side.

It is almost the same as what is done on the currenty view to inform about changes in the counters.
